util/log.py

```diff
@@ -0,0 +1,91 @@
+from __future__ import print_function
+import __builtin__
+import sys
+
+# used for looking up level from step=0 to error=4
+LEVEL_NAMES = 'step,debug,info,warn,error'.split(',')
+
+def set_log_levels(log_level_str):
+    '''
+    Sets the global log levels by parsing a log level configuration string.
+    Typically called at startup by main routine of a program (passing-in a command line value).
+    'log_level_str' - comma separated assignments of log levels to modules (e.g. "main=debug,persistence=step")
+        Modules are specified by their mudule IDs. 
+        Levels can be "step", "debug", "info", "warn" and "error".
+        If a part specifies only a level, this level will be used as the default one.
+    '''
+    global DEFAULT_LOG_LEVEL
+    global MODULE_LOG_LEVELS
```

Comment:
Is this required to be able to rely on `__builtin__` below ? It would be great if we could limit the number of global variables :)

Reply:
This is the "magic" ingredient here. It allows other modules to log under the same log level settings as set by the main program/command line driver. I think it's worth having this rather unique names to enable this.